File: app/lib/api/entity/UserEntity.dart
import 'package:json_annotation/json_annotation.dart';

part 'UserEntity.g.dart';

@JsonSerializable()
class UserEntity {
  String id;
  String phone;
  String email;
  String name;
  String surname;
  String patronymic;
  DateTime userDate;
  String identityCard;
  String? image;

  UserEntity(
      {required this.id,
        required this.phone,
        required this.email,
        required this.name,
        required this.surname,
        required this.patronymic,
        required this.userDate,
        required this.identityCard,
        this.image});

  factory UserEntity.fromJson(Map<String, dynamic> json) => _$UserEntityFromJson(json);
  Map<String, dynamic> toJson() => _$UserEntityToJson(this);
}

File: server/garant/src/main/java/com/thedeveloper/garant/entity/UserEntity.java
package com.thedeveloper.garant.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.thedeveloper.garant.entity.enums.Role;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "users")
@Getter
@Setter
public class UserEntity implements UserDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;
    @Column(unique = true)
    String phone;
    @JsonIgnore
    String password;
    String email;
    Date userDate;
    Date identityCardDay;
    String identityCard;
    String name;
    String surname;
    String patronymic;
    String image;
    @JsonIgnore
    Boolean enabled = false;
    @JsonIgnore
    Date register;
    @JsonIgnore
    @CollectionTable(name = "role", joinColumns = @JoinColumn(name = "id"))
    @Enumerated(EnumType.STRING)
    Role role;

    @JsonIgnore
    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        Set<Role> roleSet = new HashSet<>();
        roleSet.add(role);
        return roleSet;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @JsonIgnore
    @Override
    public String getUsername() {
        return phone;
    }

    @JsonIgnore
    @Override
    public boolean isAccountNonExpired() {
        return false;
    }

    @JsonIgnore
    @Override
    public boolean isAccountNonLocked() {
        return false;
    }

    @JsonIgnore
    @Override
    public boolean isCredentialsNonExpired() {
        return false;
    }

    @JsonIgnore
    @Override
    public boolean isEnabled() {
        return enabled;
    }
}
